core/grants.go

```diff
@@ -159,24 +154,15 @@ func (a *API) deleteGrant(ctx context.Context, x apiGrant) error {
 		return errors.Wrap(err)
 	}
 
-	data, err := a.raftDB.Get(ctx, GrantPrefix+x.Policy)
-	if err != nil {
-		return errors.Wrap(err)
-	}
-	// If there's nothing to delete, return success
-	if data == nil {
-		return nil
-	}
-
 	toDelete := authz.Grant{
 		GuardType: x.GuardType,
 		GuardData: guardData,
 		Protected: x.Protected, // should always be false
 	}
 
-	grantList := new(authz.GrantList)
-	err = proto.Unmarshal(data, grantList)
-	if err != nil {
+	var grantList authz.GrantList
+	found, err := a.sdb.Get(ctx, GrantPrefix+x.Policy, &grantList)
+	if err != nil || !found {
```

Comment:
When found is false, this will return `errors.Wrap(nil)`, which is `nil`, which is the right thing to do, but maybe deserves a comment?